### PR TITLE
test: add downstream check for spring-cloud-gcp

### PR DESCRIPTION
**Pull request by diegomarquezp**
Fixes https://github.com/googleapis/sdk-platform-java/issues/2534

The check takes 2 minutes

### CODE REVIEW

**Comment by diegomarquezp** [comment]
Using our [custom `setttings.xml`](https://github.com/googleapis/sdk-platform-java/blob/main/settings.xml) incurs in the following:

```
Error:      Non-resolvable import POM: The following artifacts could not be resolved: io.prometheus:prometheus-metrics-bom:pom:1.2.1 (absent): Could not find artifact io.prometheus:prometheus-metrics-bom:pom:1.2.1 in google-maven-central (https://maven-central.storage-download.googleapis.com/maven2/) @ org.springframework.boot:spring-boot-dependencies:3.3.1, /home/runner/.m2/repository/org/springframework/boot/spring-boot-dependencies/3.3.1/spring-boot-dependencies-3.3.1.pom, line 2457, column 19 -> [Help 2]
```

Removing the following line for now:
https://github.com/googleapis/sdk-platform-java/blob/240c24b21943accbbe071d33639402d0535719f1/.kokoro/presubmit/downstream-compatibility-spring.sh#L26

File: .github/workflows/downstream.yaml [hunk 1]
```diff
@@ -47,3 +47,20 @@ jobs:
         run: ./.kokoro/presubmit/common_test.sh
       - name: Perform downstream compatibility testing
         run: REPOS_UNDER_TEST="${{ matrix.repo }}" ./.kokoro/presubmit/downstream-compatibility.sh
+  downstream-compatibility-spring:
```
Comment (burkedavison) [inline .github/workflows/downstream.yaml:50]: Please update this to reflect that the scope is limited to Spring Cloud GCP generator.
```suggestion
  downstream-compatibility-spring-generator:
```

**Comment by burkedavison** [comment]
LGTM

File: .kokoro/presubmit/downstream-compatibility-spring.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  --no-transfer-progress \
+  --show-version \
+  --threads 1.5C \
+  --define maven.test.skip=true \
+  --define maven.javadoc.skip=true \
+  install
+
+
+# Generate showcase autoconfig
+bash scripts/generate-showcase.sh
```
Comment (blakeli0) [inline .kokoro/presubmit/downstream-compatibility-spring.sh:50]: Can we just do `./scripts/generate-showcase.sh`?

File: .kokoro/presubmit/downstream-compatibility-spring.sh [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+install_repo_modules
+GAPIC_GENERATOR_VERSION=$(parse_pom_version "gapic-generator-java-bom")
+echo "Install complete. gapic-generator-java-bom = $GAPIC_GENERATOR_VERSION"
+
+pushd gapic-generator-java/target
+# Download and configure spring-cloud-gcp for testing
+last_release=$(find_last_release_version "spring-cloud-gcp" "main" "GoogleCloudPlatform")
+git clone "https://github.com/GoogleCloudPlatform/spring-cloud-gcp.git" --depth=1 --branch "v$last_release"
+update_all_poms_dependency "spring-cloud-gcp" "gapic-generator-java-bom" "${GAPIC_GENERATOR_VERSION}"
```
Comment (blakeli0) [inline .kokoro/presubmit/downstream-compatibility-spring.sh:34]: @lqiu96 This `update_all_poms_dependency` looks like a nice util that maybe we can reuse in testing protoc.

**Review by blakeli0 — APPROVED** [review]

File: .kokoro/presubmit/downstream-compatibility-spring.sh (hunk 3 above)
Comment (lqiu96) [reply .kokoro/presubmit/downstream-compatibility-spring.sh:34]: The script I wrote to update the poms is pretty much a copy of this implementation. When I originally wrote it, I had the limitation where i couldn't use xmllint or xmlstartlet in kokoro.

File: .kokoro/presubmit/common.sh [hunk 4]
```diff
@@ -16,6 +16,8 @@
 # In the given directory ($1),
 #   update the pom.xml's dependency on the given artifact ($2) to the given version ($3)
 # ex: update_dependency google-cloud-java/google-cloud-jar-parent google-cloud-shared-dependencies 1.2.3
+commonScriptDir=$(realpath "$(dirname "${BASH_SOURCE[0]}")")
+
 function update_pom_dependency {
```
Comment (burkedavison) [inline .kokoro/presubmit/common.sh:21]: @lqiu96 : Please place the comment back with the function it describes:
```suggestion
commonScriptDir=$(realpath "$(dirname "${BASH_SOURCE[0]}")")

# In the given directory ($1),
#   update the pom.xml's dependency on the given artifact ($2) to the given version ($3)
# ex: update_dependency google-cloud-java/google-cloud-jar-parent google-cloud-shared-dependencies 1.2.3
function update_pom_dependency {
```

File: .kokoro/presubmit/downstream-compatibility-spring.sh (hunk 2 above)
Comment (diegomarquezp) [inline .kokoro/presubmit/downstream-compatibility-spring.sh:50]:
```suggestion
./scripts/generate-showcase.sh
```

**Pull request merged**